pkg/api/grpc_server.go

```diff
@@ -70,7 +70,15 @@ func (g *grpcCAServer) CreateSigningCertificate(ctx context.Context, request *fu
 		}
 	}
 
-	principal, err := authorize(ctx, token)
+	// Authenticate OIDC ID token by checking signature
+	idtoken, err := authorize(ctx, token)
+	if err != nil {
+		return nil, handleFulcioGRPCError(ctx, codes.Unauthenticated, err, invalidCredentials)
+	}
+	// Parse authenticated ID token into principal
+	// TODO:(nsmith5) replace this and authorize call above with
+	// just identity.IssuerPool.Authenticate()
+	principal, err := challenges.PrincipalFromIDToken(ctx, idtoken)
 	if err != nil {
 		return nil, handleFulcioGRPCError(ctx, codes.Unauthenticated, err, invalidCredentials)
```

Comment:
To maintain the same status code, we'll want to return `InvalidArgument` here - The verification of the token signature should have already happened in `authorize`.

Another error message would also be good, something like: `invalidIdentityToken = "There was an error processing the identity token."`

Reply:
Ah ok gotcha! Thanks!